src/main.rs

```diff
@@ -44,16 +48,32 @@ async fn main() -> Result<(), AstarteMessageHubError> {
     env_logger::init();
     let args = Cli::parse();
 
-    let options = MessageHubOptions::get(args.toml, args.store_directory).await?;
-    let _astarte_sdk = initialize_astarte_device_sdk(options).await?;
+    let mut msg_hub_opts = MessageHubOptions::get(args.toml, args.store_directory).await?;
 
-    //TODO add MessageHubServer and add AstarteHandler::new() on top of AstarteSDK
+    // Initailize an Astarte device
+    let device_sdk = initialize_astarte_device_sdk(&mut msg_hub_opts).await?;
+    info!("Connection to Astarte established.");
+
+    // Create a new Astarte handler
+    let handler = AstarteHandler::new(device_sdk);
+
+    // Create a new message hub
+    let message_hub = AstarteMessageHub::new(handler.clone());
+
+    // Run the protobuf server
+    let addr = format!("[::1]:{}", msg_hub_opts.grpc_socket_port)
```

Comment:
The `u32` in the `MessageHubOption` is not a `u16` only because the library we use to parse `.toml` files does not support it.
It's used here:
https://github.com/astarte-platform/astarte-message-hub/blob/168c83b396d640b2677491af81818aecbcc26cda/src/config/file.rs#L44-L50
Feel free to use the suggestions from github for such one liner changes. It makes it quicker to commit them.

Reply:
FYR as we talked offline, here a patch with the change since it not possible to suggest changes on other files

<details><summary>Details</summary>

```diff
diff --git a/src/config/http.rs b/src/config/http.rs
index 07d12d7..6047a0b 100644
--- a/src/config/http.rs
+++ b/src/config/http.rs
@@ -56,7 +56,7 @@ struct ConfigPayload {
     credentials_secret: Option<String>,
     pairing_url: String,
     pairing_token: Option<String>,
-    grpc_socket_port: u32,
+    grpc_socket_port: u16,
 }
 
 impl Default for ConfigResponse {
diff --git a/src/config/mod.rs b/src/config/mod.rs
index 4bcc5c6..c643839 100644
--- a/src/config/mod.rs
+++ b/src/config/mod.rs
@@ -42,7 +42,7 @@ pub struct MessageHubOptions {
     pub pairing_token: Option<String>,
     pub interfaces_directory: Option<String>,
     pub astarte_ignore_ssl: bool,
-    pub grpc_socket_port: u32,
+    pub grpc_socket_port: u16,
 }
 
 impl MessageHubOptions {
@@ -107,7 +107,6 @@ impl MessageHubOptions {
             && (valid_secret || valid_token)
             && valid_interface_dir
             && (!self.pairing_url.is_empty())
-            && (self.grpc_socket_port <= 655356)
     }
 }
 
@@ -220,21 +219,6 @@ mod test {
         assert!(!expected_msg_hub_opts.is_valid());
     }
 
-    #[test]
-    fn test_is_valid_invalid_grpc_socket_port_err() {
-        let expected_msg_hub_opts = MessageHubOptions {
-            realm: "1".to_string(),
-            device_id: "2".to_string(),
-            pairing_url: "3".to_string(),
-            credentials_secret: None,
-            pairing_token: Some("4".to_string()),
-            interfaces_directory: None,
-            astarte_ignore_ssl: false,
-            grpc_socket_port: 655357,
-        };
-        assert!(!expected_msg_hub_opts.is_valid());
-    }
-
     #[test]
     fn test_is_valid_invalid_interf_dir_err() {
         let expected_msg_hub_opts = MessageHubOptions {
diff --git a/src/config/protobuf.rs b/src/config/protobuf.rs
index 046f41d..1a81626 100644
--- a/src/config/protobuf.rs
+++ b/src/config/protobuf.rs
@@ -19,6 +19,7 @@
  */
 
 use std::io::Write;
+use std::num::TryFromIntError;
 use std::path::Path;
 
 use tokio::sync::mpsc::{channel, Sender};
@@ -46,6 +47,15 @@ impl proto_message_hub::message_hub_config_server::MessageHubConfig for AstarteM
         request: Request<proto_message_hub::ConfigMessage>,
     ) -> Result<Response<pbjson_types::Empty>, Status> {
         let req = request.into_inner();
+
+        // Protobuf version 3 only supports u32
+        let port: u16 = req
+            .grpc_socket_port
+            .try_into()
+            .map_err(|err: TryFromIntError| {
+                Status::new(Code::InvalidArgument, format!("Invalid grpc port: {}", err))
+            })?;
+
         let message_hub_options = MessageHubOptions {
             realm: req.realm,
             device_id: req.device_id,
@@ -54,7 +64,7 @@ impl proto_message_hub::message_hub_config_server::MessageHubConfig for AstarteM
             pairing_token: req.pairing_token,
             interfaces_directory: None,
             astarte_ignore_ssl: false,
-            grpc_socket_port: req.grpc_socket_port,
+            grpc_socket_port: port,
         };
 
         if !message_hub_options.is_valid() {
diff --git a/src/main.rs b/src/main.rs
index bfb2d62..107aa30 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,3 +1,5 @@
+use std::net::Ipv6Addr;
+
 /*
  * This file is part of Astarte.
  *
@@ -61,10 +63,10 @@ async fn main() -> Result<(), AstarteMessageHubError> {
     let message_hub = AstarteMessageHub::new(handler.clone());
 
     // Run the protobuf server
-    let socket: std::net::SocketAddr = (std::net::Ipv6Addr::LOCALHOST, 8080).into();
+    let addrs = (Ipv6Addr::LOCALHOST, msg_hub_opts.grpc_socket_port).into();
     tonic::transport::Server::builder()
         .add_service(MessageHubServer::new(message_hub))
-        .serve(socket)
+        .serve(addrs)
         .await?;
 
     Ok(())

```

</details>